File: Challenges/21 Days DSA Challenge/Insert node recursively.cpp
/****************************************************************

    Following is the class structure of the Node class:

        class Node
        {
        public:
            int data;
            Node *next;
            Node(int data)
            {
                this->data = data;
                this->next = NULL;
            }
        };

*****************************************************************/
Node *InsertNode(Node *head, int pos, Node *newNode)
{
    if (pos == 0)
    {
        newNode->next = head;
        return newNode;
    }

    if (head == nullptr)
    {
        return nullptr;
    }

    head->next = InsertNode(head->next, pos - 1, newNode);
    return head;
}

File: Challenges/21 Days DSA Challenge/Insert node recursively.py
'''
Following is the class structure of Node Class

class Node:
    def __init__(data):
        self.data = data
        self.next = None
'''


def InsertNode(head, pos, newNode):
    if pos == 0:
        newNode.next = head
        return newNode

    if head is None:
        return None

    head.next = InsertNode(head.next, pos - 1, newNode)
    return head
